solara/components/input.py

```diff
@@ -386,7 +391,7 @@ def on_external_value_change(new_value: Optional[T]):
 
         return reactive_value.subscribe(on_external_value_change)
 
-    solara.use_effect(sync_back_input_value, [reactive_value])
+    solara.use_effect(sync_back_input_value, [reactive_value, *extra_dependencies])
```

Comment:
This should be `reactive_value.value` and maybe instead of extra_dependencies, use stringify.

I think our equals is good enough to compare separate function which have the same closure and code to be the same.

Reply:
I think not using `reactive_value.value` is correct here since the effect subscribes to changes on `reactive_value`. Since the `reactive_value` is declared at hook level I don't think having it in the dependencies ever causes a rerun of the effect, so we could remove it for clarity?

Reply:
Ah true!